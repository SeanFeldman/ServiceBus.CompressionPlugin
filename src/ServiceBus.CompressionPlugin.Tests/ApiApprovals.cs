namespace ServiceBus.CompressionPlugin.Tests
{
    using PublicApiGenerator;
    using ServiceBus.CompressionPlugin;
    using Xunit;

    public class ApiApprovals
    {
        [Fact]
        public void CompressionPlugin()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(CompressionPlugin).Assembly,
                new ApiGeneratorOptions
                {
                    WhitelistedNamespacePrefixes = new[] { "Microsoft.Azure.ServiceBus." },
                    ExcludeAttributes = new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" }
                });

            Approver.Verify(publicApi);
        }
    }
}