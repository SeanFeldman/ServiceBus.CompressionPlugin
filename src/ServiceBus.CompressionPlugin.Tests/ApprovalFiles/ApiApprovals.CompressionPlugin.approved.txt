[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("ServiceBus.CompressionPlugin.Tests")]
namespace Microsoft.Azure.ServiceBus
{
    public class CompressionConfiguration
    {
        public CompressionConfiguration(string compressionMethodName, System.Func<byte[], byte[]> compressor, int minimumSize, System.Collections.Generic.Dictionary<string, System.Func<byte[], byte[]>> decompressors) { }
        public CompressionConfiguration(string compressionMethodName, System.Func<byte[], byte[]> compressor, int minimumSize, System.Func<byte[], byte[]> decompressor) { }
        public void AddDecompressor(string compressionMethodName, System.Func<byte[], byte[]> decompressor) { }
    }
    public static class CompressionPluginExtensions
    {
        public static Microsoft.Azure.ServiceBus.Core.ServiceBusPlugin RegisterCompressionPlugin(this Microsoft.Azure.ServiceBus.ClientEntity client, Microsoft.Azure.ServiceBus.CompressionConfiguration compressionConfiguration) { }
        public static Microsoft.Azure.ServiceBus.Core.ServiceBusPlugin RegisterCompressionPlugin(this Microsoft.Azure.ServiceBus.ClientEntity client, int minimumSizeToApplyCompression = 1500) { }
    }
}